pupepat/utils.py

```diff
@@ -68,14 +127,63 @@ def save_results(input_filename: str, best_fit_models: list,
             best_fit_model['filename'] = os.path.basename(input_filename)
             for keyword in header_keywords:
                 best_fit_model[keyword] = hdu[0].header[keyword]
-            logger.info(best_fit_model)
+            logger.debug(best_fit_model)
             output_table.add_row(best_fit_model)
 
     output_table.write(output_filename, format='ascii', overwrite=True)
     return output_table
 
 
+def source_is_valid(data, s):
+    """
+    Validate source as reasonable. Checks:
+    1. bad columns in image
+    2. source is in focus (not a pupil)
+    3. too close to edge of image
+    4. not circular enough
+
+    :param data: image data from which the source was extracted
+    :param s: SEP extracted source
+
+    :return Boolean is_valid
+    """
+    bad_column_max = config['source_filters']['bad_column_max']
+    bad_column_min = config['source_filters']['bad_column_min']
+    got_bad_columns = (s['xmax'] - s['xmin']) < bad_column_min and (s['ymax'] - s['ymin']) > bad_column_max
+    got_bad_columns |= (s['xmax'] - s['xmin']) > bad_column_max and (s['ymax'] - s['ymin']) < bad_column_min
+
+    focus_scale_factor = config['source_filters']['focus_scale_factor']
+    background = np.median(data)
+    in_focus = np.abs(data[int(s['y']), int(s['x'])] - background) > focus_scale_factor * np.sqrt(background)
```

Comment:
I agree with the confusing name (and I'm open to suggestions :-) I added the parameterization stuff, but otherwise the logic has not changed. let's talk if you want to re-visit this code.